IonHashDotNet/IonHasher.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace Amazon.IonHash
+{
+    using System.Security.Cryptography;
+
+    public class IonHasher : IIonHasher
```

Comment:
let's rename this class to something like DefaultIonHasher, CryptoIonHasher, DotnetIonhasher, or ...?  Some name that hints at the underlying implementation (and thereby what algorithms are available)

Reply:
note also that instances of IIonHasher may be reused by the Ion Hash implementation, but it appears that HashAlgorithm implementations are apparently [not guaranteed to be reusable](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.hashalgorithm.canreusetransform?view=netframework-4.8#System_Security_Cryptography_HashAlgorithm_CanReuseTransform)

Reply:
Renamed to CryptoIonHasher

Reply:
`HashAlgorithm.CanReuseTransform` is always `true`, so my second comment above is incorrect; no changes needed